### PR TITLE
feat(quotes): BACK-600 quote draft shipping expectation

### PR DESCRIPTION
**Pull request by declankirk**
Jira: [BACK-600](https://bigcommercecloud.atlassian.net/browse/BACK-600)

## What/Why?
Adding the shipping expectation message to the quote draft view. Also introducing `draftQuoteListHasBackorderedItemsForDisplay` function to centralise the “any qualifying backorder line?” logic for `QuoteSummary` and `QuoteTable`.

## Rollout/Rollback
Backorder display is gated by `Feature_Backorder && BACK-134.backorders_phase_1_1_control_messaging_on_storefront`.

## Testing
Shipping expectation message appears for quotes with any backordered quantity.

https://github.com/user-attachments/assets/7de92ea5-2269-4d01-ac5c-12f7f566c737

Shipping expectation message does not appear for disabled `ShowDefaultShippingExpectationPrompt`.

https://github.com/user-attachments/assets/92cc2736-2d96-4cb0-ad19-92d3cb353703

Shipping expectation message does not appear for disabled `Feature_Backorder` or `BACK-134.backorders_phase_1_1_control_messaging_on_storefront`.

https://github.com/user-attachments/assets/189bcc57-3cab-43e2-8818-5976fd948f1b

[BACK-600]: https://bigcommercecloud.atlassian.net/browse/BACK-600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

> [!NOTE]
> **Low Risk**
> Low risk UI/UX change gated behind existing backorder settings and a feature flag, plus test coverage; the main risk is incorrect backorder-qualification logic affecting prompt/toggle visibility.
> 
> **Overview**
> Adds a shipping expectation message to the quote draft `QuoteSummary`, shown only when **backorders are enabled**, the storefront messaging flag `BACK-134.backorders_phase_1_1_control_messaging_on_storefront` is on, the default prompt setting is enabled, and the draft contains at least one backordered line.
> 
> Centralizes “any qualifying backorder line?” detection in the new utility `draftQuoteListHasBackorderedItemsForDisplay`, reusing it in both `QuoteSummary` and `QuoteTable` (including hiding the "Backorder details" toggle when messaging is disabled). Adds unit tests covering the new prompt/toggle gating and the helper’s edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1c0f8bcca8e74f8a6dc7b84cd99c68b76bdbb375. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/storefront/src/pages/quote/utils/getDraftBackorderDisplayFields.ts [hunk 1]
```diff
@@ -107,3 +107,11 @@ export function getDraftBackorderDisplayFields(
     quantityBackordered,
   };
 }
+
+export function draftQuoteListHasBackorderedItemsForDisplay(draftQuoteList: QuoteItem[]): boolean {
+  return draftQuoteList.some((quoteItem) => {
+    const fields = getDraftBackorderDisplayFields(quoteItem.node);
+    if (!fields) return false;
+    return !draftRowQuantityExceedsAvailableToSell(quoteItem.node);
```
Comment (benpratt77) [inline apps/storefront/src/pages/quote/utils/getDraftBackorderDisplayFields.ts:115]: is this meant to be inverted?
seems like we return true if  draftRowQuantityExceedsAvailableToSell returns false.

Comment (declankirk) [reply apps/storefront/src/pages/quote/utils/getDraftBackorderDisplayFields.ts:115]: `draftRowQuantityExceedsAvailableToSell` is quantity validation - i.e. when `qty > ATS` we show an error, not the backorder prompts.

Comment (benpratt77) [reply apps/storefront/src/pages/quote/utils/getDraftBackorderDisplayFields.ts:115]: ah.... good call, completely in the prompt space lol

File: apps/storefront/src/pages/quote/components/QuoteSummary.test.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        },
+      },
+    } as QuoteItem;
+
+    renderWithProviders(<QuoteSummary />, withPromptEnabled([lineInStock]));
+
+    expect(screen.queryByText(expectationMessage)).toBeNull();
+  });
+
+  it('hides the prompt when storefront messaging feature flag is off, even with backordered lines', () => {
```
Comment (benpratt77) [inline apps/storefront/src/pages/quote/components/QuoteSummary.test.tsx]: items sounds better not lines. 
Or perhaps lineItems

Same above a few times I see line

File: apps/storefront/src/pages/quote/utils/getDraftBackorderDisplayFields.test.ts [hunk 3]
```diff
@@ -186,3 +187,54 @@ describe('draftRowQuantityExceedsAvailableToSell', () => {
     expect(draftRowQuantityExceedsAvailableToSell(row)).toBe(true);
   });
 });
+
+describe('draftQuoteListHasBackorderedItemsForDisplay', () => {
+  it('is false for an empty list', () => {
+    expect(draftQuoteListHasBackorderedItemsForDisplay([])).toBe(false);
+  });
+
+  it('is true when a line has backorder fields and quantity is within available-to-sell', () => {
```
Comment (benpratt77) [inline apps/storefront/src/pages/quote/utils/getDraftBackorderDisplayFields.test.ts]: Lets say `returns` rather is `true`
Ad also `line` here.

File: apps/storefront/src/pages/quote/utils/getDraftBackorderDisplayFields.test.ts [hunk 4]
```diff
@@ -186,3 +187,54 @@ describe('draftRowQuantityExceedsAvailableToSell', () => {
...
+        totalOnHand: 3,
+        availableToSell: 10,
+        unlimitedBackorder: false,
+      },
+    } as QuoteLineNode;
+
+    expect(draftQuoteListHasBackorderedItemsForDisplay([{ node: row }] as QuoteItem[])).toBe(true);
+  });
+
+  it('is false when ordered quantity is within total on hand (no backordered quantity)', () => {
```
Comment (benpratt77) [inline apps/storefront/src/pages/quote/utils/getDraftBackorderDisplayFields.test.ts]: Same here, `returns false..`

File: apps/storefront/src/pages/quote/components/QuoteSummary.tsx [hunk 5]
```diff
@@ -40,6 +45,20 @@ const QuoteSummary = forwardRef((_, ref: Ref<unknown>) => {
...
   const draftQuoteList = useAppSelector(({ quoteInfo }) => quoteInfo.draftQuoteList);
+  const backorderEnabled = useIsBackorderEnabled();
+  const isBackorderMessagingEnabled = useFeatureFlag(
+    'BACK-134.backorders_phase_1_1_control_messaging_on_storefront',
+  );
+  const { showDefaultShippingExpectationPrompt, defaultShippingExpectationPrompt } = useAppSelector(
+    ({ global }) => global.backorderDisplaySettings,
+  );
+
+  const hasBackorderedItems = useMemo(() => {
```
Comment (benpratt77) [inline apps/storefront/src/pages/quote/components/QuoteSummary.tsx:56]: This feels a bit strange.

The `isBackorderMessagingEnabled` flag still controls the hiding and showing of the prompt. but we are using it control the list of backroderedItems.

Is this more saving us a call since we won't need that data here anyway?

Comment (declankirk) [reply apps/storefront/src/pages/quote/components/QuoteSummary.tsx:56]: Yeah this is from a Copilot recommendation above, short-circuiting an unnecessary call.

https://github.com/bigcommerce/b2b-buyer-portal/pull/837#discussion_r3120992840

Comment (benpratt77) [reply apps/storefront/src/pages/quote/components/QuoteSummary.tsx:56]: Fair call, this will be removed in time anyway.

File: apps/storefront/src/pages/quote/components/QuoteSummary.tsx [hunk 6]
```diff
@@ -40,6 +45,20 @@ const QuoteSummary = forwardRef((_, ref: Ref<unknown>) => {
...
+    if (!isBackorderMessagingEnabled) {
+      return false;
+    }
+    return draftQuoteListHasBackorderedItemsForDisplay(draftQuoteList);
+  }, [draftQuoteList, isBackorderMessagingEnabled]);
```
Comment (bc-gary) [inline apps/storefront/src/pages/quote/components/QuoteSummary.tsx:61]: **[Suggestion]** Minor asymmetry: this `useMemo` guards with `isBackorderMessagingEnabled` to skip list iteration when the flag is off (nice perf touch), but `QuoteTable` calls `draftQuoteListHasBackorderedItemsForDisplay(items)` without this guard.

Both are correct since downstream checks handle it either way. Just flagging the inconsistency — consider aligning the two if you want the pattern to be uniform for future readers.

File: apps/storefront/src/pages/quote/components/QuoteSummary.test.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+          backorderMessage: 'Restock soon',
+        },
+      },
+    } as QuoteItem;
+
+    renderWithProviders(<QuoteSummary />, withMessagingFlagDisabled([lineItemWithBackorder]));
+
+    expect(screen.queryByText(expectationMessage)).toBeNull();
+  });
+});
```
Comment (bc-gary) [inline apps/storefront/src/pages/quote/components/QuoteSummary.test.tsx]: **[Suggestion]** Good coverage of the three main scenarios. One gap: there is no test for `backorderEnabled: false`. The `ShippingExpectationPrompt` component returns null when `backorderEnabled` is false, and that IS tested in `ShippingExpectationPrompt.test.tsx` — so no actual coverage gap. But adding a fourth case here (e.g., `withBackorderDisabled`) would make this integration suite self-documenting.

**Review by bc-gary — APPROVED** [review]
Clean PR — good extraction, proper feature gating, consistent patterns with QuoteDetailSummary. Two minor suggestions inline, no blockers.

**Review by benpratt77 — APPROVED** [review]

File: apps/storefront/src/pages/quote/components/QuoteSummary.test.tsx [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+      },
+    }),
+    quoteInfo: buildQuoteInfoWithList(draftQuoteList),
+  },
+});
+
+describe('QuoteSummary shipping expectation prompt', () => {
+  it('shows the prompt when the draft has backordered items and messaging is enabled', () => {
+    const lineItemWithBackorder: QuoteItem = {
+      node: {
```
Comment (bc-gauravsobti) [inline apps/storefront/src/pages/quote/components/QuoteSummary.test.tsx]: I see this lineItemWithBackorder repeating in 4 tests. Best to extract it out. This will shorten this file and also make sure that this pattern is not repeated. 
I see similar pattern in other test files for mock api response, many times it's repeated, and then later on it becomes unmanageable(developers think that this duplication might have been done because of some reason)

File: apps/storefront/src/pages/quote/components/QuoteSummary.tsx [hunk 9]
```diff
@@ -141,6 +160,15 @@ const QuoteSummary = forwardRef((_, ref: Ref<unknown>) => {
               </Typography>
             </Grid>
 
+            {isBackorderMessagingEnabled && (
+              <ShippingExpectationPrompt
```
Comment (bc-gauravsobti) [inline apps/storefront/src/pages/quote/components/QuoteSummary.tsx:164]: [Question]
Since all this props can be fetched from redux using hooks, isn't it better to avoid passing props? This way we are making sure context is inside one component. What do you think?

File: apps/storefront/src/pages/quote/components/QuoteTable.tsx [hunk 10]
```diff
@@ -186,15 +187,12 @@ function QuoteTable({ total, items, updateSummary }: QuoteTableProps) {
...
-  const hasBackorderedItems = useMemo(
-    () =>
-      items.some((quoteItem) => {
-        const fields = getDraftBackorderDisplayFields(quoteItem.node);
-        if (!fields) return false;
-        return !draftRowQuantityExceedsAvailableToSell(quoteItem.node);
-      }),
-    [items],
-  );
+  const hasBackorderedItems = useMemo(() => {
```
Comment (bc-gauravsobti) [inline apps/storefront/src/pages/quote/components/QuoteTable.tsx:190]: [Question]
Is `hasBackorderedItems` being used in many places? Does it provide value to extract into its separate file?

File: apps/storefront/src/pages/quote/components/QuoteSummary.tsx [hunk 11]
```diff
@@ -141,6 +160,15 @@ const QuoteSummary = forwardRef((_, ref: Ref<unknown>) => {
...
+                backorderEnabled={backorderEnabled}
+                hasBackorderedItems={hasBackorderedItems}
+                showDefaultShippingExpectationPrompt={showDefaultShippingExpectationPrompt}
```
Comment (bc-jeffteng) [inline apps/storefront/src/pages/quote/components/QuoteSummary.tsx:167]: [nit]
Test cover the scenario when `showDefaultShippingExpectationPrompt` is `false` ?

File: apps/storefront/src/pages/quote/components/QuoteTable.tsx [hunk 12]
```diff
@@ -186,15 +187,12 @@ function QuoteTable({ total, items, updateSummary }: QuoteTableProps) {
...
+    if (!isBackorderMessagingEnabled) {
+      return false;
+    }
+    return draftQuoteListHasBackorderedItemsForDisplay(items);
+  }, [items, isBackorderMessagingEnabled]);
```
Comment (bc-jeffteng) [inline apps/storefront/src/pages/quote/components/QuoteTable.tsx:195]: [Question]
Do we have test cover this scenario?

**Review by bc-jeffteng — APPROVED** [review]
Nice PR. I only have two comments, but they’re not blockers, so I’ll approve

File: apps/storefront/src/pages/quote/components/QuoteSummary.tsx (hunk 9 above)
Comment (declankirk) [reply apps/storefront/src/pages/quote/components/QuoteSummary.tsx:164]: `hasBackorderedItems` is derived from the draft list here and passed from the parent on quote detail, so the child would still need props or duplicate logic.

File: apps/storefront/src/pages/quote/components/QuoteTable.tsx (hunk 10 above)
Comment (declankirk) [reply apps/storefront/src/pages/quote/components/QuoteTable.tsx:190]: `hasBackorderedItems` is only used in `QuoteTable` to derive `showBackorderToggle`, the shared logic is already in `draftQuoteListHasBackorderedItemsForDisplay`.

**Review by bc-gauravsobti — APPROVED** [review]
LGTM
